dove/tests/helper.rs

```diff
@@ -0,0 +1,181 @@
+    dove_toml_path.push("Dove.toml");
+    let mut toml_value = read_to_string(&dove_toml_path)
+        .unwrap()
+        .parse::<Value>()
+        .unwrap();
+    let mut dd = toml::map::Map::new();
+    dd.insert(
+        "path".to_string(),
+        Value::String(move_stdlib.to_str().unwrap().to_string()),
+    );
+    toml_value
+        .get_mut("package")
+        .unwrap()
+        .as_table_mut()
+        .unwrap()
+        .insert(
+            "dependencies".to_string(),
+            Value::Array(vec![Value::Table(dd)]),
+        );
+    write_all(
+        &dove_toml_path,
+        toml::to_string(&toml_value).unwrap().as_str(),
+    )
+    .unwrap();
+}
+
+pub fn execute_dove_at(args: &[&str], project_folder: &Path) -> anyhow::Result<()> {
+    execute(args, project_folder.to_path_buf())
+}
+pub fn execute_dove_bin_at(args: &[&str], project_folder: &Path) -> anyhow::Result<String> {
+    assert!(project_folder.exists());
+    let output = std::process::Command::new(env!("CARGO_BIN_EXE_dove"))
+        .current_dir(project_folder)
+        .args(args.iter().skip(1)) // TODO: remove first argument `dove` on callee side
+        .output()?;
+    anyhow::ensure!(
+        output.status.success(),
+        "Command {:?} failed with code {}",
+        args,
+        output.status
+    );
+    Ok(String::from_utf8(output.stdout)?)
+}
+
+pub fn check_dove_toml(
+    project_folder: &Path,
+    project_name: &str,
+    dialect: Option<&str>,
+    address: Option<&str>,
+    api: Option<&str>,
+) -> Result<(), Error> {
+    // Check config
+    let dove_toml_string = read_to_string(project_folder.join("Dove.toml"))
+        .unwrap()
+        .replace(" ", "")
+        .replace("\t", "");
+
+    ensure!(
+        dove_toml_string.contains(&format!("name=\"{}\"", project_name)),
+        "Dove.toml: invalid name\n".to_string()
+    );
+    if let Some(dialect) = dialect {
+        ensure!(
+            dove_toml_string.contains(&format!("dialect=\"{}\"", dialect)),
+            "Dove.toml: invalid dialect"
+        );
+    }
+    if let Some(address) = address {
+        ensure!(
+            dove_toml_string.contains(&format!("account_address=\"{}\"", address)),
+            "Dove.toml: invalid account_address\n"
+        );
+    }
+    if let Some(api) = api {
+        ensure!(
+            dove_toml_string.contains(&format!("blockchain_api=\"{}\"", api)),
+            "Dove.toml: invalid blockchain_api\n"
+        );
+    }
+
+    Ok(())
+}
+
+pub fn check_for_base_categories(project_folder: &Path) -> Result<(), Error> {
+    ensure!(
+        project_folder.join("modules").exists(),
+        "Modules folder: not found\n".to_string()
+    );
+    ensure!(
+        project_folder.join("scripts").exists(),
+        "Scripts folder: not found\n".to_string()
+    );
+    ensure!(
+        project_folder.join("scripts").exists(),
+        "Scripts folder: not found\n".to_string()
+    );
+
+    Ok(())
```

Comment:
You're using this function with `.unwrap()` everywhere, with more obvious rename you can hide it inside the function. `check_` prefix suggests that it will assert from inside. 
You'll have to replace `ensure!` with `assert!` here.